login/service.go

```diff
@@ -357,18 +357,24 @@ func (keycloak *KeycloakOAuthProvider) AuthCodeCallback(ctx *app.CallbackAuthori
 		return nil, err
 	}
 	var redirectTo string
-	parameters := referrerURL.Query()
-	parameters.Add("code", ctx.Code)
-	parameters.Add("state", ctx.State)
+
+	var parameters url.Values
 
 	if responseMode != nil && *responseMode == "fragment" {
```

Comment:
wouldn't it be simpler to move the logic of checking the response mode to build the referrerURL in a separate function, that could be exhaustively tested in a simpler manner ?

Reply:
addressed in https://github.com/fabric8-services/fabric8-auth/pull/697/commits/d8e19d215298611f8b070df7fd4d96acbe866c52